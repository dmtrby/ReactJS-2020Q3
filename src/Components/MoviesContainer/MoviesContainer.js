import React from 'react';
import PropTypes from 'prop-types';

import './movies-container.scss';
import MovieCard from '../MovieCard';

const MoviesContainer = ({ movies }) => (
  <div className="container">
    <div className="row flex-column">
      <div className="col-xs-12">
        <span>
          <strong>{movies.length}</strong>
          {' '}
          movies found
        </span>
      </div>
      <div className="col-xs-12 col-no-gutter margin-bottom-6">
        <div className="row">
          {movies.map((film) => <div className="col-xs-12 col-md-6 col-lg-6 col-xl-4 col-no-gutter-lr" key={film.id}><MovieCard {...film} /></div>)}
        </div>
      </div>
    </div>
  </div>
);

MoviesContainer.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default MoviesContainer;
