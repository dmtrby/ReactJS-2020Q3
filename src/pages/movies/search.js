import React, { useEffect } from 'react';
import { useRouter } from 'next/router';
import { useDispatch } from 'react-redux';
import { fetchMovies } from '../../redux/actions';
import Head from 'next/head';

import MainPageHeader from '../../Components/MainPageHeader';
import HeaderSection from '../../Components/HeaderSection';
import FooterSection from '../../Components/FooterSection';
import Main from '../../Components/Main';

// import sprite from './icons/icons-sprite.svg';

function MoviesPage({ noFetch }) {
  const dispatch = useDispatch();

  useEffect(() => {
    if (!noFetch) {
      const router = useRouter();
      const {
        query: { filter, sortBy, search },
      } = router;
      dispatch(fetchMovies(filter, sortBy, search));
    }
  }, [dispatch]);

  return (
    <>
      <Head>
        <title>Movies gallery | Netflix roulette</title>
        <meta name="description" content="Movies gallery page" />
      </Head>
      <HeaderSection>
        <MainPageHeader></MainPageHeader>
      </HeaderSection>
      <Main isEmpty={false} />
      <FooterSection />
    </>
  );
}

MoviesPage.getInitialProps = async ({ store, query }) => {
  const { filter, sortBy, search } = query;
  await store.dispatch(fetchMovies(filter, sortBy, search));
  return {
    noFetch: true,
  };
};

export default MoviesPage;
