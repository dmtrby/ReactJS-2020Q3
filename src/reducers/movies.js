const initialMoviesState = {
  movies: [],
  isLoading: false,
  error: null,
  isDetailsPage: false,
  detailsMovieData: null,
  filter: 'All',
  sort: 'title',
  needToUpdate: true,
};

const movies = (state = initialMoviesState, action) => {
  switch (action.type) {
    case 'FETCH_MOVIES_BEGIN':
      return {
        ...state,
        isLoading: true,
        error: null,
        movies: [],
      };
    case 'FETCH_MOVIES_SUCCESS':
      return {
        ...state,
        isLoading: false,
        movies: action.payload.movies,
        needToUpdate: false,
      };
    case 'ADD_MOVIE_SUCCESS':
      return {
        ...state,
        needToUpdate: true,
      };
    case 'DELETE_MOVIE_SUCCESS':
      return {
        ...state,
        needToUpdate: true,
      };
    case 'DELETE_MOVIE_FAILURE':
      return {
        ...state,
        needToUpdate: true,
      };
    case 'EDIT_MOVIE_SUCCESS':
      return {
        ...state,
        needToUpdate: true,
      };
    case 'CHANGE_FILTER':
      return {
        ...state,
        filter: action.filter,
      };
    case 'CHANGE_SORT':
      return {
        ...state,
        sort: action.sort,
      };
    case 'SET_MAIN_PAGE':
      return {
        ...state,
        isDetailsPage: false,
        detailsMovieData: null,
      };
    case 'SET_DETAILS_PAGE':
      return {
        ...state,
        isDetailsPage: true,
        detailsMovieData: state.movies.find((v) => v.id === action.id),
      };
    default:
      return state;
  }
};

export default movies;
