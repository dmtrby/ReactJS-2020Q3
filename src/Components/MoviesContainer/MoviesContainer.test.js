import React from 'react';
import MoviesContainer from './MoviesContainer';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { initialMoviesState } from '../../redux/reducers/movies';

const middleWares = [thunk];
const mockStore = configureMockStore(middleWares);
const store = mockStore(initialMoviesState);

const mockedMoviesData = [
  {
    genres: ['Action', 'Adventure', 'Fantasy'],
    id: '1',
    overview: 'overviewText',
    poster_path: 'https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg',
    release_date: '2017-10-25',
    runtime: 130,
    title: 'Thor: Ragnarok',
  },
  {
    genres: ['Action', 'Horror', 'Fantasy'],
    id: '2',
    overview: 'overviewText',
    poster_path: 'https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg',
    release_date: '2017-10-25',
    runtime: 130,
    title: 'Aviator',
  },
  {
    genres: ['Action', 'Adventure'],
    id: '3',
    overview: 'overviewText',
    poster_path: 'https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg',
    release_date: '2017-10-25',
    runtime: 130,
    title: 'ScarFace',
  },
];

store.movies = mockedMoviesData;

test('MoviesContainer should matches the Snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <MoviesContainer movies={mockedMoviesData} />
      </Router>
    </Provider>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
