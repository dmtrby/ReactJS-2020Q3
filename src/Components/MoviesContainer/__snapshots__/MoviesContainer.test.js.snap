// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should matches the Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row flex-column"
  >
    <div
      className="col-xs-12"
    >
      <span>
        <strong>
          3
        </strong>
        movies found
      </span>
    </div>
    <div
      className="col-xs-12 col-no-gutter margin-bottom-6"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-md-6 col-lg-6 col-xl-4 col-no-gutter-lr"
        >
          <article
            className="movie-card"
          >
            <button
              aria-controls="menu--1"
              aria-haspopup={true}
              className="button button--text movie-card__menu"
              data-reach-menu-button=""
              id="menu-button--menu"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              <svg
                className="icon  icon--light"
              >
                <use
                  xlinkHref="#icons-sprite_more"
                />
              </svg>
            </button>
            <div
              data-reach-menu=""
              data-reach-menu-popover=""
              hidden={true}
            >
              <div
                aria-labelledby="menu-button--menu"
                data-reach-menu-items=""
                data-reach-menu-list=""
                id="menu--1"
                onKeyDown={[Function]}
                role="menu"
                tabIndex={-1}
              >
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Edit
                </div>
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Remove
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <a
                  href="/film/1"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    className="movie-card__img"
                    src="https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg"
                  />
                </a>
              </div>
              <div
                className="col-xs-12 padding-bottom-0 padding-top-0"
              >
                <div
                  className="row between-xs"
                >
                  <h3
                    className=""
                  >
                    Thor: Ragnarok
                  </h3>
                  <div
                    className="movie-card__year"
                  >
                    2017
                  </div>
                </div>
              </div>
              <div
                className="col-xs-12 padding-top-0"
              >
                <span
                  className="movie-card__genre"
                >
                  Action
                  , 
                </span>
                <span
                  className="movie-card__genre"
                >
                  Adventure
                  , 
                </span>
                <span
                  className="movie-card__genre"
                >
                  Fantasy
                  
                </span>
              </div>
            </div>
          </article>
        </div>
        <div
          className="col-xs-12 col-md-6 col-lg-6 col-xl-4 col-no-gutter-lr"
        >
          <article
            className="movie-card"
          >
            <button
              aria-controls="menu--2"
              aria-haspopup={true}
              className="button button--text movie-card__menu"
              data-reach-menu-button=""
              id="menu-button--menu"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              <svg
                className="icon  icon--light"
              >
                <use
                  xlinkHref="#icons-sprite_more"
                />
              </svg>
            </button>
            <div
              data-reach-menu=""
              data-reach-menu-popover=""
              hidden={true}
            >
              <div
                aria-labelledby="menu-button--menu"
                data-reach-menu-items=""
                data-reach-menu-list=""
                id="menu--2"
                onKeyDown={[Function]}
                role="menu"
                tabIndex={-1}
              >
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Edit
                </div>
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Remove
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <a
                  href="/film/2"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    className="movie-card__img"
                    src="https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg"
                  />
                </a>
              </div>
              <div
                className="col-xs-12 padding-bottom-0 padding-top-0"
              >
                <div
                  className="row between-xs"
                >
                  <h3
                    className=""
                  >
                    Aviator
                  </h3>
                  <div
                    className="movie-card__year"
                  >
                    2017
                  </div>
                </div>
              </div>
              <div
                className="col-xs-12 padding-top-0"
              >
                <span
                  className="movie-card__genre"
                >
                  Action
                  , 
                </span>
                <span
                  className="movie-card__genre"
                >
                  Horror
                  , 
                </span>
                <span
                  className="movie-card__genre"
                >
                  Fantasy
                  
                </span>
              </div>
            </div>
          </article>
        </div>
        <div
          className="col-xs-12 col-md-6 col-lg-6 col-xl-4 col-no-gutter-lr"
        >
          <article
            className="movie-card"
          >
            <button
              aria-controls="menu--3"
              aria-haspopup={true}
              className="button button--text movie-card__menu"
              data-reach-menu-button=""
              id="menu-button--menu"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              <svg
                className="icon  icon--light"
              >
                <use
                  xlinkHref="#icons-sprite_more"
                />
              </svg>
            </button>
            <div
              data-reach-menu=""
              data-reach-menu-popover=""
              hidden={true}
            >
              <div
                aria-labelledby="menu-button--menu"
                data-reach-menu-items=""
                data-reach-menu-list=""
                id="menu--3"
                onKeyDown={[Function]}
                role="menu"
                tabIndex={-1}
              >
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Edit
                </div>
                <div
                  className="button button--text movie-card__menu-item"
                  data-reach-menu-item=""
                  data-selected=""
                  data-valuetext=""
                  onClick={[Function]}
                  onDragStart={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseMove={[Function]}
                  onMouseUp={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  Remove
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <a
                  href="/film/3"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    className="movie-card__img"
                    src="https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg"
                  />
                </a>
              </div>
              <div
                className="col-xs-12 padding-bottom-0 padding-top-0"
              >
                <div
                  className="row between-xs"
                >
                  <h3
                    className=""
                  >
                    ScarFace
                  </h3>
                  <div
                    className="movie-card__year"
                  >
                    2017
                  </div>
                </div>
              </div>
              <div
                className="col-xs-12 padding-top-0"
              >
                <span
                  className="movie-card__genre"
                >
                  Action
                  , 
                </span>
                <span
                  className="movie-card__genre"
                >
                  Adventure
                  
                </span>
              </div>
            </div>
          </article>
        </div>
      </div>
    </div>
  </div>
</div>
`;
